libs/ast_generic/AST_utils.ml

```diff
@@ -40,26 +40,57 @@ module Node_ID = Gensym.MkId ()
    so as to select structural or referential equality.
 *)
 
-type busy_with_equal = Not_busy | Structural_equal | Referential_equal
+type busy_with_equal =
+  | Not_busy
+  | Structural_equal
+  | Referential_equal
+  | Syntactic_equal
```

Comment:
you need to add a comment 4 lines above explaining what is a syntactic equal and why we need yet another form of equality.